src/Symfony/Component/Security/Core/Authorization/Voter/AbstractVoter.php

```diff
@@ -107,7 +120,30 @@ public function vote(TokenInterface $token, $object, array $attributes)
      * @param object               $object
      * @param UserInterface|string $user
      *
+     * @deprecated Override voteOnAttribute instead
+     *
+     * @return bool
+     */
+    protected function isGranted($attribute, $object, $user = null)
+    {
+        return false;
+    }
+
+    /**
+     * Perform a single access check operation on a given attribute, object and (optionally) user
+     * It is safe to assume that $attribute and $object's class pass supportsAttribute/supportsClass
+     * $user can be one of the following:
+     *   a UserInterface object (fully authenticated user)
+     *   a string               (anonymously authenticated user).
+     *
+     * @param string         $attribute
+     * @param object         $object
+     * @param TokenInterface $token
+     *
```

Comment:
Can you add a note that this method should be abstract in 3.0 so that we do not forget that when cleaning the master branch?